test/org/sagebionetworks/bridge/validators/ConsentSignatureValidatorTest.java

```diff
@@ -0,0 +1,240 @@
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageData\":\"\",\n" +
+                "   \"imageMimeType\":\"image/fake\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        assertValidatorMessage(validator, sig, "imageData", "cannot be an empty string");
+    }
+
+    @Test
+    public void jsonEmptyImageMimeType() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageData\":\"" + TestConstants.DUMMY_IMAGE_DATA + "\",\n" +
+                "   \"imageMimeType\":\"\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        assertValidatorMessage(validator, sig, "imageMimeType", "cannot be an empty string");
+    }
+
+    @Test
+    public void jsonImageDataWithoutMimeType() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageData\":\"" + TestConstants.DUMMY_IMAGE_DATA + "\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void jsonImageMimeTypeWithoutData() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageMimeType\":\"image/fake\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void minAgeLimitButNoBirthdate() {
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").build();
+        assertValidatorMessage(validator, sig, "birthdate", "cannot be missing, null, or blank");
+    }
+    
+    @Test
+    public void minAgeLimitButBirthdateTooRecent() {
+        String birthdate = new DateTime(SIGNED_ON_TIMESTAMP).minusYears(18).plusDays(1).toLocalDate().toString();
```

Comment:
This test will fail on Feb 28 2022. See https://sagebionetworks.jira.com/browse/BRIDGE-2136 (And possibly the opposite test will fail on Feb 29 2020. The tests in ConsentServiceTest currently using plusDays(2) for this exact reason.

I dunno if this code will still exist in 2-4 years time, but I think we should future-proof the test if not the code.